app/routes/goal_routes.py

```diff
@@ -1 +1,118 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from ..db import db
+from ..models.goal import Goal
+from ..models.task import Task
+from .route_utilities import validate_model
+from datetime import datetime, timezone
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+@goals_bp.post("")
+def create_goal():
+
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except KeyError as e:
+        if e.args[0] == "title":
+            response = {"details": "Invalid data"}
+        return make_response(response, 400)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response = {"goal": new_goal.create_response()}
+
+    return response, 201
+
+
+@goals_bp.post("/<goal_id>/tasks")
+def create_tasks_with_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    task_ids = request_body["task_ids"]
+    for task_id in task_ids:
+        task = validate_model(Task, task_id)
+        task.goal_id = goal.id
+
+        db.session.commit()
+
+    response = {"id": goal.id}
+    response.update(request_body)
```

Comment:
Line 45 assumes that `request_body` will only have one set of key/value pairs and the are `task_ids` and a list of ids. We can't make this assumption though because we don't know what a client will send to us. If a client sent a request body that had `task_id`, `title`, `other_random_stuff` then when you call `response.update(request_body)` on line 45, you will create a response body with all the random information the client passed along. This is one reason we don't want to call `update` on response with `request_cody`. 

Another reason is we don't want to take the client-provided task ids and just return it back in our response, doing so would be like just echoing back what the client told us. Instead, it would be nice to return a response that has task ids that we get from the goal that we fetched from the db on line 33 that way we can ensure that these task ids actually exist in our source of truth (our DB).

You'd need to grab the task ids from the goal like:
```suggestion
response = {"id": goal.id}
tasks_from_goal = goal.tasks
task_ids_from_goal = [task.id for task in tasks_from_goal]

response["task_ids"] = task_ids_from_goal
```